pkg/operator/client/deploy.go

```diff
@@ -216,38 +218,55 @@ func (c *Client) ensureResourcesPresent(deployArgs operatortypes.DeployAppArgs)
 	return &deployRes, nil
 }
 
-func (c *Client) installWithHelm(helmDir string, kotsCharts []v1beta1.HelmChart) (*commandResult, error) {
-	version := "3"
-	chartsDir := filepath.Join(helmDir, "charts")
-
-	orderedDirs, err := getSortedCharts(chartsDir, kotsCharts, c.TargetNamespace, false)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to get sorted helm charts")
-	}
+func (c *Client) installWithHelm(v1Beta1ChartsDir, v1beta2ChartsDir string, kotsCharts []kotsutil.HelmChartInterface) (*commandResult, error) {
+	// sort the charts by weight and then by name
+	sort.Slice(kotsCharts, func(i, j int) bool {
+		if kotsCharts[i].GetWeight() != kotsCharts[j].GetWeight() {
+			return kotsCharts[i].GetWeight() < kotsCharts[j].GetWeight()
+		}
+		return kotsCharts[i].GetChartName() < kotsCharts[j].GetChartName()
+	})
 
+	version := "3"
 	var hasErr bool
 	var multiStdout, multiStderr [][]byte
-	for _, dir := range orderedDirs {
-		installDir := filepath.Join(chartsDir, dir.Name)
-		args := []string{"upgrade", "-i", dir.ReleaseName, installDir, "--timeout", "3600s"}
 
-		if dir.Namespace != "" {
-			args = append(args, "-n", dir.Namespace)
+	for _, kotsChart := range kotsCharts {
+		var args []string
+		if kotsChart.GetAPIVersion() == "kots.io/v1beta1" {
+			installDir := filepath.Join(v1Beta1ChartsDir, kotsChart.GetDirName())
+			args = []string{"upgrade", "-i", kotsChart.GetReleaseName(), installDir, "--timeout", "3600s"}
+		} else if kotsChart.GetAPIVersion() == "kots.io/v1beta2" {
+			installDir := filepath.Join(v1beta2ChartsDir, kotsChart.GetDirName())
+			chartPath := filepath.Join(installDir, fmt.Sprintf("%s-%s.tgz", kotsChart.GetChartName(), kotsChart.GetChartVersion()))
+			valuesPath := filepath.Join(installDir, "values.yaml")
+			args = []string{"upgrade", "-i", kotsChart.GetReleaseName(), chartPath, "-f", valuesPath, "--timeout", "3600s"}
```

Comment:
how about separating out the shared args? e.g.: move `"upgrade", "-i", kotsChart.GetReleaseName()` to the array initialization, and the `"--timeout", "3600s"` to afterwards